tutorials/xarray_tutorial.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) 2018 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+xarray with MetPy Tutorial
+==========================
+
+`xarray <http://xarray.pydata.org/>`_ is a powerful Python package that provides N-dimensional
+labeled arrays and datasets following the Common Data Model. While the process of integrating
+xarray features into MetPy is ongoing, this tutorial demonstrates how xarray can be used
+within the current version of MetPy. MetPy's integration primarily works through accessors
+which allow simplified projection handling and coordinate identification. Unit and calculation
+support is currently available in a limited fashion, but should be improved in future
+versions.
+"""
+
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import matplotlib.pyplot as plt
+import xarray as xr
+
+# Any import of metpy will activate the accessors
+import metpy.calc as mpcalc
+from metpy.testing import get_test_data
+from metpy.units import units
+
+#########################################################################
+# Getting Data
+# ------------
+#
+# While xarray can handle a wide variety of n-dimensional data (essentially anything that can
+# be stored in a netCDF file), a common use case is working with model output. Such model
+# data can be obtained from a THREDDS Data Server using the siphon package, but for this
+# tutorial, we will use an example subset of GFS data from Hurrican Irma (September 5th,
+# 2017).
+
+# Open the netCDF file as a xarray Dataset
+data = xr.open_dataset(get_test_data('irma_gfs_example.nc', False))
+
+# View a summary of the Dataset
+print(data)
+
+#########################################################################
+# Preparing Data
+# --------------
+#
+# To make use of the data within MetPy, we need to parse the dataset for projection and
+# coordinate information following the CF conventions. For this, we use the
+# ``data.metpy.parse_cf()`` method.
+#
+# Additionally, we take a few optional steps to clean up our dataset for later use:
+#
+# - Drop the extra grid mapping variable since we have already parsed that information
```

Comment:
I'm not clear on why we're going to the effort of dropping a variable. Does it help with printing or something?

Reply:
Good point...I was being over-zealous in my clean up of the Dataset.